bgpd/bgp_mplsvpn.c

```diff
@@ -1270,6 +1316,165 @@ void vpn_policy_routemap_event(const char *rmap_name)
 		vpn_policy_routemap_update(bgp, rmap_name);
 }
 
+void vrf_import_from_vrf(struct bgp *bgp, struct bgp *vrf_bgp,
+			 afi_t afi, safi_t safi)
+{
+	const char *export_name;
+	vpn_policy_direction_t idir, edir;
+	char *vname;
+	char buf[1000];
+	struct ecommunity *ecom;
+	bool first_export = false;
+
+	export_name = bgp->name ? bgp->name : VRF_DEFAULT_NAME;
+	idir = BGP_VPN_POLICY_DIR_FROMVPN;
+	edir = BGP_VPN_POLICY_DIR_TOVPN;
+
+	/*
+	 * Cross-ref both VRFs. Also, note if this is the first time
+	 * any VRF is importing from "import_vrf".
+	 */
+	vname = XSTRDUP(MTYPE_TMP, vrf_bgp->name);
```

Comment:
Can `vrf_bgp->name` ever be NULL? Does it need the same treatment with VRF_DEFAULT_NAME as `export_name` above?

Reply:
you are correct, when I dove deeper into this I noticed I had missed some commits.  Which I am adding to the PR.